### ISSUE TITLE
A way to configure forward proxy settings for injected sidecars

### ISSUE THREAD

**Issue by jsmichaels** [linkerd/linkerd2#3352]
## Feature Request
There should be a way to configure forward proxy settings for the injected sidecars.

### What problem are you trying to solve?
We're trying to get Linkerd working in one of our production clusters, which is behind our corporate firewall. Anything that wants to get to the public internet must use our forward proxy. The linkerd-proxy sidecar doesn't use this, so applications which have it injected into their pod are unable to reach the internet.

### How should the problem be solved?
We need a way to configure the linkerd-proxy sidecar to use our forward proxy server, either via environment variables (HTTPS_PROXY etc) or some other mechanism. This configuration should allow for NO_PROXY settings as well.

### Any alternatives you've considered?
We could change the firewall rules on our production cluster to allow all internet traffic out. This is very undesirable.

### How would users interact with this feature?
This could be passed to the linkerd CLI tool by parameters. Something like:
`linkerd install --forward-proxy-https=my-proxy.corp.net --forward-proxy-noproxy=127.0.0.1,10.0.0.0/8,.corp.net,.svc | kubectl apply -f -`

**Comment by jsmichaels** [comment]
Any update on this?

**Comment by grampelberg** [comment]
@jsmichaels we'd love help working on it, there is no one currently working on it that I'm aware of.

**Comment by javaducky** [comment]
I could maybe take a stab at this @grampelberg .  Feels like an `iptables` hack, would you agree?

**Comment by grampelberg** [comment]
Hmmm, I had the assumption we'd need to do this in the proxy. If we just have the proxy forward *everything* to the egress, it'll catch internal traffic too. Ideally that would be gated by the HOST header. Now that I'm thinking about it though, we'll have the original IP address and can switch between traffic originally destined for inside the cluster or outside it. We're kinda doing something similar there inside the proxy itself for the TCP mTLS work. 

WDYT about throwing a POC together and generating some conversation? I'm uncomfortable that iptables is the correct implementation and a POC would let us see how well it fits together with the other pieces.

**Comment by javaducky** [comment]
Agreed...my first reaction was `iptables` but yeah...needs something a bit higher-level and in the proxy I'm sure.
